node/taskcommand.ts

```diff
@@ -33,12 +34,16 @@ export class TaskCommand {
                     var val = this.properties[key];
                     if (val) {
                         cmdStr += key + '=' + val + ';';
+                        if(key === 'secret' && val === 'true') {
+                            containsSecret = true;
+                        }
                     }                    
                 }
             }
         }
 
-        cmdStr += ']' + this.message;
+        cmdStr += ']';
+        cmdStr += containsSecret? '********' : this.message;
```

Comment:
OK - I see why because you call cmd and it debug logs cmd tostring.  So once again you're trying to register the secret but the act of registering the secret is logging the secret.